Re-enable plugin and upgrade bats tests

rjernst opened the pull request:
While the tests remained in place, the execution of these tests were
lost when creating the DistroTestPlugin. This commit restores executing
them.

rjernst committed:
Re-enable plugin and upgrade bats tests

While the tests remained in place, the execution of these tests were
lost when creating the DistroTestPlugin. This commit restores executing
them.

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Packaging)

mark-vieira commented on qa/os/build.gradle:
```diff
@@ -58,7 +58,7 @@ tasks.dependenciesInfo.enabled = false
 
 tasks.register('destructivePackagingTest') {
-  dependsOn 'destructiveDistroTest' 
+  dependsOn 'destructiveDistroTest', 'destructiveBatsTest.plugins', 'destructiveBatsTest.upgrade'
```
Would simply using `tasks.withType(BatsTestTask)` work just as well here and be slightly less brittle?

rjernst replied:
While it would work here, it would not work in the later example because only the destructive test is of type BatsTestTask. The latter is a GradleDistroTestTask, which is a wrapper to call into the VM. Given only these two cases remain, and their time should be limited, I'm not sure it is worth further trying to generify this.

mark-vieira commented on qa/os/build.gradle:
```diff
@@ -67,7 +67,7 @@ processTestResources {
 subprojects { Project platformProject ->
   tasks.register('packagingTest') {
-    dependsOn 'distroTest', 'batsTest.oss', 'batsTest.default'
+    dependsOn 'distroTest', 'batsTest.plugins', 'batsTest.upgrade'
```
Same comment as above.

mark-vieira reviewed (approved):
One comment regarding making this a bit less fragile. Otherwise LGTM.

rjernst committed:
fix line length

rjernst committed:
auto format...

rjernst committed:
Merge branch 'master' into distro_tests41

rjernst committed:
Merge branch 'master' into distro_tests41

rjernst committed:
Merge branch 'master' into distro_tests41

rjernst committed:
debug code...

rjernst committed:
Close stdout/stderr before replacing with logging stream

We replace the System.out and System.err objects with redirects to our
log file, but never ensure they are closed. This commit closes each
stream just before replacing it.

rjernst committed:
try another

rjernst committed:
Merge branch 'master' into distro_tests41

rjernst committed:
cleanup

rjernst committed:
Merge branch 'master' into logging6

rjernst committed:
Merge branch 'master' into distro_tests41

rjernst committed:
Merge branch 'logging6' into distro_tests41

rjernst committed:
test cleanup

rjernst committed:
can't check asm jar

rjernst committed:
Merge branch 'master' into distro_tests41

rjernst committed:
relative path

pugnascotia commented:
Can we get this finished off and merged? It's causing a lot of build failues that need triaged.

rjernst committed:
upgrade debug

rjernst commented:
@pugnascotia I have been working getting these tests working all week. Unfortunately there were many things broken over the months since they were accidentally disabled. There is still one remaining failure that happens on certain versions with the upgrade tests, which I am still looking into. I pushed 37210c4 to fix #51974, which I did not realize was an issue.

rjernst committed:
Merge branch 'master' into distro_tests41

rjernst committed:
Merge branch 'master' into distro_tests41

rjernst committed:
Do not daemonize when testing error output in packaging tests

The packaging tests start elasticsearch in various ways. All of these
currently expect it is started asynchronously, yet some tests expect it
will fail to start and want to check the error output. This commit adds
a daemonize flag to the utility methods to start elasticsearch for such
cases, so that when the start method returns, all the error output
should already be available since the process will have exited.

relates #51716

rjernst committed:
checkstyle

rjernst committed:
clear windows log file after test

rjernst committed:
spotless

rjernst committed:
Merge branch 'distro_tests45' into distro_tests41
